### ISSUE TITLE
net-wireless/rtl8821ce-driver-5.5.2_p20211119: fails to compile

### ISSUE THREAD

**Issue by ghost** [dallenwilson/trolltoo#7]
Hi there, I've tried emerging it but that's the result: any ideas?
```
 * Package:    net-wireless/rtl8821ce-driver-5.5.2_p20211119
 * Repository: trolltoo
 * Maintainer: dwjwilson@lavabit.com
 * USE:        abi_x86_64 amd64 elibc_glibc kernel_linux userland_GNU
 * FEATURES:   network-sandbox preserve-libs sandbox userpriv usersandbox
 * Determining the location of the kernel source code
 * Found kernel source directory:
 *     /usr/src/linux
 * Found sources for kernel version:
 *     5.15.19-gentoo-dist
 * Checking for suitable kernel configuration options...                                                                                                                                                                             [ ok ]
>>> Unpacking source...
>>> Unpacking ca204c60724d23ab10244f920d4e50759ed1affb.tar.gz to /var/tmp/portage/net-wireless/rtl8821ce-driver-5.5.2_p20211119/work
>>> Source unpacked in /var/tmp/portage/net-wireless/rtl8821ce-driver-5.5.2_p20211119/work
>>> Preparing source in /var/tmp/portage/net-wireless/rtl8821ce-driver-5.5.2_p20211119/work/rtl8821ce-ca204c60724d23ab10244f920d4e50759ed1affb ...
 * Applying use-correct-kernel.patch ...                                                                                                                                                                                             [ ok ]
>>> Source prepared.
>>> Configuring source in /var/tmp/portage/net-wireless/rtl8821ce-driver-5.5.2_p20211119/work/rtl8821ce-ca204c60724d23ab10244f920d4e50759ed1affb ...
>>> Source configured.
>>> Compiling source in /var/tmp/portage/net-wireless/rtl8821ce-driver-5.5.2_p20211119/work/rtl8821ce-ca204c60724d23ab10244f920d4e50759ed1affb ...
ln: failed to create symbolic link 'Module.symvers': File exists
 * Preparing 8821ce module
make -j5 HOSTCC=x86_64-pc-linux-gnu-gcc 'LDFLAGS=-m elf_x86_64' clean modules 
#make -C /lib/modules/5.15.19-gentoo-dist/build M=/var/tmp/portage/net-wireless/rtl8821ce-driver-5.5.2_p20211119/work/rtl8821ce-ca204c60724d23ab10244f920d4e50759ed1affb clean
make ARCH=x86 CROSS_COMPILE=x86_64-pc-linux-gnu- -C /lib/modules/5.15.19-gentoo-dist/build M=/var/tmp/portage/net-wireless/rtl8821ce-driver-5.5.2_p20211119/work/rtl8821ce-ca204c60724d23ab10244f920d4e50759ed1affb  modules
cd hal ; rm -fr */*/*/*.mod.c */*/*/*.mod */*/*/*.o */*/*/.*.cmd */*/*/*.ko
make[1]: Entering directory '/usr/src/linux-5.15.19-gentoo-dist'
cd hal ; rm -fr */*/*.mod.c */*/*.mod */*/*.o */*/.*.cmd */*/*.ko
cd hal ; rm -fr */*.mod.c */*.mod */*.o */.*.cmd */*.ko
cd hal ; rm -fr *.mod.c *.mod *.o .*.cmd *.ko
cd core ; rm -fr */*.mod.c */*.mod */*.o */.*.cmd */*.ko
cd core ; rm -fr *.mod.c *.mod *.o .*.cmd *.ko
cd os_dep/linux ; rm -fr *.mod.c *.mod *.o .*.cmd *.ko
cd os_dep ; rm -fr *.mod.c *.mod *.o .*.cmd *.ko
cd platform ; rm -fr *.mod.c *.mod *.o .*.cmd *.ko
rm -fr Module.symvers ; rm -fr Module.markers ; rm -fr modules.order
rm -fr *.mod.c *.mod *.o .*.cmd *.ko *~
rm -fr .tmp_versions
  CC [M]  /var/tmp/portage/net-wireless/rtl8821ce-driver-5.5.2_p20211119/work/rtl8821ce-ca204c60724d23ab10244f920d4e50759ed1affb/core/rtw_cmd.o
  CC [M]  /var/tmp/portage/net-wireless/rtl8821ce-driver-5.5.2_p20211119/work/rtl8821ce-ca204c60724d23ab10244f920d4e50759ed1affb/core/rtw_security.o
  CC [M]  /var/tmp/portage/net-wireless/rtl8821ce-driver-5.5.2_p20211119/work/rtl8821ce-ca204c60724d23ab10244f920d4e50759ed1affb/core/rtw_debug.o
  CC [M]  /var/tmp/portage/net-wireless/rtl8821ce-driver-5.5.2_p20211119/work/rtl8821ce-ca204c60724d23ab10244f920d4e50759ed1affb/core/rtw_io.o
  CC [M]  /var/tmp/portage/net-wireless/rtl8821ce-driver-5.5.2_p20211119/work/rtl8821ce-ca204c60724d23ab10244f920d4e50759ed1affb/core/rtw_ioctl_query.o
error: objtool: check subcommand not implemented
make[2]: *** [scripts/Makefile.build:277: /var/tmp/portage/net-wireless/rtl8821ce-driver-5.5.2_p20211119/work/rtl8821ce-ca204c60724d23ab10244f920d4e50759ed1affb/core/rtw_ioctl_query.o] Error 38
make[2]: *** Deleting file '/var/tmp/portage/net-wireless/rtl8821ce-driver-5.5.2_p20211119/work/rtl8821ce-ca204c60724d23ab10244f920d4e50759ed1affb/core/rtw_ioctl_query.o'
make[2]: *** Waiting for unfinished jobs....
error: objtool: check subcommand not implemented
make[2]: *** [scripts/Makefile.build:277: /var/tmp/portage/net-wireless/rtl8821ce-driver-5.5.2_p20211119/work/rtl8821ce-ca204c60724d23ab10244f920d4e50759ed1affb/core/rtw_io.o] Error 38
make[2]: *** Deleting file '/var/tmp/portage/net-wireless/rtl8821ce-driver-5.5.2_p20211119/work/rtl8821ce-ca204c60724d23ab10244f920d4e50759ed1affb/core/rtw_io.o'
error: objtool: check subcommand not implemented
make[2]: *** [scripts/Makefile.build:277: /var/tmp/portage/net-wireless/rtl8821ce-driver-5.5.2_p20211119/work/rtl8821ce-ca204c60724d23ab10244f920d4e50759ed1affb/core/rtw_security.o] Error 38
make[2]: *** Deleting file '/var/tmp/portage/net-wireless/rtl8821ce-driver-5.5.2_p20211119/work/rtl8821ce-ca204c60724d23ab10244f920d4e50759ed1affb/core/rtw_security.o'
error: objtool: check subcommand not implemented
make[2]: *** [scripts/Makefile.build:277: /var/tmp/portage/net-wireless/rtl8821ce-driver-5.5.2_p20211119/work/rtl8821ce-ca204c60724d23ab10244f920d4e50759ed1affb/core/rtw_cmd.o] Error 38
make[2]: *** Deleting file '/var/tmp/portage/net-wireless/rtl8821ce-driver-5.5.2_p20211119/work/rtl8821ce-ca204c60724d23ab10244f920d4e50759ed1affb/core/rtw_cmd.o'
error: objtool: check subcommand not implemented
make[2]: *** [scripts/Makefile.build:277: /var/tmp/portage/net-wireless/rtl8821ce-driver-5.5.2_p20211119/work/rtl8821ce-ca204c60724d23ab10244f920d4e50759ed1affb/core/rtw_debug.o] Error 38
make[2]: *** Deleting file '/var/tmp/portage/net-wireless/rtl8821ce-driver-5.5.2_p20211119/work/rtl8821ce-ca204c60724d23ab10244f920d4e50759ed1affb/core/rtw_debug.o'
make[1]: *** [Makefile:1868: /var/tmp/portage/net-wireless/rtl8821ce-driver-5.5.2_p20211119/work/rtl8821ce-ca204c60724d23ab10244f920d4e50759ed1affb] Error 2
make[1]: Leaving directory '/usr/src/linux-5.15.19-gentoo-dist'
make: *** [Makefile:2244: modules] Error 2
 * ERROR: net-wireless/rtl8821ce-driver-5.5.2_p20211119::trolltoo failed (compile phase):
 *   emake failed
 * 
 * If you need support, post the output of `emerge --info '=net-wireless/rtl8821ce-driver-5.5.2_p20211119::trolltoo'`,
 * the complete build log and the output of `emerge -pqv '=net-wireless/rtl8821ce-driver-5.5.2_p20211119::trolltoo'`.
 * The complete build log is located at '/var/tmp/portage/net-wireless/rtl8821ce-driver-5.5.2_p20211119/temp/build.log'.
 * The ebuild environment file is located at '/var/tmp/portage/net-wireless/rtl8821ce-driver-5.5.2_p20211119/temp/environment'.
 * Working directory: '/var/tmp/portage/net-wireless/rtl8821ce-driver-5.5.2_p20211119/work/rtl8821ce-ca204c60724d23ab10244f920d4e50759ed1affb'
 * S: '/var/tmp/portage/net-wireless/rtl8821ce-driver-5.5.2_p20211119/work/rtl8821ce-ca204c60724d23ab10244f920d4e50759ed1affb'
```

**Comment by dallenwilson** [comment]
I haven't seen anything like that before. Are you using sys-kernel/gentoo-sources, or another kernel?

A few thoughts:

- Have you tried the in-kernel rtw88 module? It supports some of the 8821ce chipsets, but not all of them. I keep this ebuild around because so far mine is one of the unsupported ones.
- In your log snippet, `make ARCH=x86 CROSS_COMPILE=x86_64-pc-linux-gnu` jumped out at me. I've never tested any cross-compiling with this ebuild. I've also never tested it for an x86 machine, just amd64. The driver may not work at all for an x86 machine.
- Lastly, there are two other ebuilds for this driver (possibly pointed at the same upstream git commit, I haven't checked), in the guru and calculate overlays. The guru overlay tends to have well-tested ebuilds. Might be worth trying that one.

**Comment by ghost** [comment]
Hi, I'm using gentoo-kernel-bin 5.16.5 (while writing it was the 5.15.19). I don't really like the rtw88 driver (I have the rtl8821ce chipset)

**Comment by dallenwilson** [comment]
I installed the gentoo-kernel-bin package, eselect'd that kernel, and rebuilt this module successfully, so I don't know what to suggest. Try building it manually against the upstream git repo, I guess.

**Comment by karimrir1** [comment]
@dallenwilson Got a completely different error on 5.17, However there is a pull request on the rtl8821ce repo that will fix this. it isn't merged yet, The link to it is https://github.com/tomaspinho/rtl8821ce/pull/267

**Comment by karimrir1** [comment]
> Hi, I'm using gentoo-kernel-bin 5.16.5 (while writing it was the 5.15.19). I don't really like the rtw88 driver (I have the rtl8821ce chipset)

Works fine here on the 5.16.16 gentoo-source packages, however not the 5.17.0 one
Edit: Typo

**Comment by dallenwilson** [comment]
When upstream merges the fixes for 5.17, I'll update the ebuild here to use it. I'm not much interested in maintaining the driver itself or any patches to it.
